libcoap/src/types.rs

```diff
@@ -322,12 +322,14 @@ impl From<CoapProtocol> for CoapUriScheme {
 /// # Note on URI Length Limits
 ///
 /// Due to [the specified limits](https://datatracker.ietf.org/doc/html/rfc7252#section-5.10)
-/// of CoAP option lengths, URI path components for a request/location URI must not exceed 255
-/// bytes each, i.e. a full path with more than 255 bytes is fine, but each individual path segment
-/// must be smaller than 255 bytes.
+/// of CoAP option lengths, URI path components, query components and hostnames for a URI must not
```

Comment:
Oxford comma and a small clarification.

```suggestion
/// of CoAP option lengths, the URI path components, query components, and hostnames for a URI must not
```